src/forms/FormCheckboxInput.d.ts

```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+
+declare namespace FormCheckboxInput {
+    interface IFormCheckboxInputProps<T> extends React.HTMLProps<T> {
+      label: string;
+    }
+    type Props = IFormCheckboxInputProps<HTMLElement>; // TODO: not sure if `Form.Group`
+    interface IFormCheckboxInputState {
+      value: boolean;
+    }
+    type State = IFormCheckboxInputState;
+    type Component = React.Component<Props, State>;
+}
+declare const FormCheckboxInput: FormCheckboxInput.Component; // TODO: Should this be a class?
```

Comment:
It's fine, you can use `class` to keep it consistent with the api since it doesn't have nested components but this would work just fine.